### ISSUE TITLE
Crash with insight version 4.0.4

### ISSUE THREAD

**Issue by hzhangxyz** [penguin0616/insight#34]
log is here: https://paste.ubuntu.com/p/6FQ5GxfPgS/

**Comment by hzhangxyz** [comment]
Seems in `deerclopsspawner.lua:22`, filename is set to nil by mistake?

**Comment by penguin0616** [comment]
Do any of these mods modify Deerclops or its spawning? This issue doesn't happen when I use just Insight.

**Comment by hzhangxyz** [comment]
I did not add any mod to modifying thing about deerclops explicitly, I dont know whether some mod changed it without telling me.

I add a `mprint(filename)` before the crash point, and it shows filename is nil. What is this variable expected to be?

**Comment by penguin0616** [comment]
It should be ``"deerclopsspawner"``.
What do you get if you put 
```lua
mprint(debug.getinfo(1, "S").source)
```
before the crash point? Or after the variable declaration?

**Comment by hzhangxyz** [comment]
I add `mprint("hzhangxyz:0:", debug.getinfo(1, "S").source)` and `mprint("hzhangxyz:1:", filename)` after the line `local filename = debug.getinfo(1, "S").source:match("([%w_]+)%.lua$")`

Then `tail -n 1000000 -f server_log.txt | grep hzhangxyz:` got:
```
[00:04:10]: [workshop-2189004162 (Insight)]:    hzhangxyz:0:    ../mods/workshop-2189004162/scripts/descriptors/deerclopsspawner.lua
[00:04:10]: [workshop-2189004162 (Insight)]:    hzhangxyz:1:    nil
```

**Comment by hzhangxyz** [comment]
I did not use lua before, but this behavior really confuses me:
```
Lua 5.4.4  Copyright (C) 1994-2022 Lua.org, PUC-Rio
> local a = 1
> print(a)
nil
> 
```

**Comment by hzhangxyz** [comment]
I change the code to this
```
-- deerclopsspawner.lua [Worldly]
local filename = debug.getinfo(1, "S").source:match("([%w_]+)%.lua$")
mprint("hzhangxyz:0:", debug.getinfo(1, "S").source)
mprint("hzhangxyz:1:", filename)
mprint("hzhangxyz:2:", debug.getinfo(1, "S").source:match("([%w_]+)%.lua$"))
```

Then, the result of `tail -n 1000000 -f server_log.txt | grep hzhangxyz:` is
```
[00:03:18]: [workshop-2189004162 (Insight)]:    hzhangxyz:0:    ../mods/workshop-2189004162/scripts/descriptors/deerclopsspawner.lua
[00:03:18]: [workshop-2189004162 (Insight)]:    hzhangxyz:1:    nil
[00:03:18]: [workshop-2189004162 (Insight)]:    hzhangxyz:2:    nil
```

So it is not the fault of that behavior:


> I did not use lua before, but this behavior really confuses me:
> 
> ```
> Lua 5.4.4  Copyright (C) 1994-2022 Lua.org, PUC-Rio
> > local a = 1
> > print(a)
> nil
> > 
> ```

**Comment by penguin0616** [comment]
> I did not use lua before, but this behavior really confuses me:
> 
> ```
> Lua 5.4.4  Copyright (C) 1994-2022 Lua.org, PUC-Rio
> > local a = 1
> > print(a)
> nil
> > 
> ```

Since that variable was declared locally, it only exists within the scope of that executed statement.
Doing 
```lua
local a=1 print(a)
```
in one statement would print 1.

---

> I add `mprint("hzhangxyz:0:", debug.getinfo(1, "S").source)` and `mprint("hzhangxyz:1:", filename)` after the line `local filename = debug.getinfo(1, "S").source:match("([%w_]+)%.lua$")`
> 
> Then `tail -n 1000000 -f server_log.txt | grep hzhangxyz:` got:
> 
> ```
> [00:04:10]: [workshop-2189004162 (Insight)]:    hzhangxyz:0:    ../mods/workshop-2189004162/scripts/descriptors/deerclopsspawner.lua
> [00:04:10]: [workshop-2189004162 (Insight)]:    hzhangxyz:1:    nil
> ```

Okay, so that tells me that there's a mod messing with how strings get processed in game.
Usually, the culprit is a Chinese translation mod. Are you using one of those?

**Comment by hzhangxyz** [comment]
Yes, I added a mod with workshop id 1301033176 , which is about chinese translation.

**Comment by penguin0616** [comment]
> Yes, I added a mod with workshop id 1301033176 , which is about chinese translation.

Thought so.
Can you crash again, and provide the full log this time? There's some debug information in the log I'll need that should tell me why Insight's not properly compensating for that mod.

**Comment by hzhangxyz** [comment]
I try to disable that mod and the server stop crashing. Thank you very much.

The full log when crashing is here: https://paste.ubuntu.com/p/fBRH46pF2F/

**Comment by penguin0616** [comment]
Thanks for the log.
I released an update (4.0.5) that should hopefully fix the crash when you have that mod enabled.

Does it work for you?
